src/v/wasm/cache.cc

```diff
@@ -317,24 +323,27 @@ ss::future<ss::shared_ptr<factory>> caching_runtime::make_factory(
     co_return cached;
 }
 
-ss::future<> caching_runtime::do_gc() {
+ss::future<int64_t> caching_runtime::do_gc() {
```

Comment:
here and elsewhere, would there be any advantage (or disadvantage) to using a strong type?

Reply:
I personally don't think so. This value is exposed only for testing and is not exposed in a public interface.

Philosophically, this is a count (of GC'd entries) and doesn't have semantic meaning, so generally we haven't used strong types from what I've seen. Things like offsets, ids, etc all have semantic meaning and benefit conceptually from the strong type, but for a simple count like this I don't see an added benefit.

Reply:
In general I'm very pro named_type however.

Reply:
Ya, I misread this as some kind of error tracking thing. I guess it was late 😅